src/playercomponent.js

```diff
@@ -190,46 +190,48 @@ function PlayerComponent (playbackElement, bigscreenPlayerData, mediaSources, wi
     publishMediaStateUpdate(undefined, { timeUpdate: true })
   }
 
-  function onError () {
+  function onError (mediaError) {
     bubbleBufferingCleared()
-    raiseError()
+    raiseError(mediaError)
   }
 
   function startBufferingErrorTimeout () {
     const bufferingTimeout = isInitialPlay ? 30000 : 20000
     clearBufferingErrorTimeout()
     errorTimeoutID = setTimeout(() => {
       bubbleBufferingCleared()
-      attemptCdnFailover(true)
+      attemptCdnFailover(true, {code: PluginEnums.ERROR_CODES.BUFFERING_TIMEOUT, message: PluginEnums.ERROR_MESSAGES.BUFFERING_TIMEOUT})
     }, bufferingTimeout)
   }
 
-  function raiseError () {
+  function raiseError (mediaError) {
     clearBufferingErrorTimeout()
     publishMediaStateUpdate(MediaState.WAITING)
-    bubbleErrorRaised()
-    startFatalErrorTimeout()
+    bubbleErrorRaised(mediaError)
+    startFatalErrorTimeout(mediaError)
   }
 
-  function startFatalErrorTimeout () {
+  function startFatalErrorTimeout (mediaError) {
     if (!fatalErrorTimeout && !fatalError) {
       fatalErrorTimeout = setTimeout(() => {
         fatalErrorTimeout = null
         fatalError = true
-        attemptCdnFailover(false)
+        attemptCdnFailover(false, mediaError)
       }, 5000)
     }
   }
 
-  function attemptCdnFailover (bufferingTimeoutError) {
+  function attemptCdnFailover (bufferingTimeoutError, mediaError) {
```

Comment:
This could use the mediaError code instead of the separate `bufferingTimeoutError` argument.